x-pack/plugin/ql/src/main/java/org/elasticsearch/xpack/ql/execution/search/extractor/AbstractFieldHitExtractor.java

```diff
@@ -81,16 +90,15 @@ public void writeTo(StreamOutput out) throws IOException {
     @Override
     public Object extract(SearchHit hit) {
         Object value = null;
-        DocumentField field = null;
         if (hitName != null) {
-            value = unwrapFieldsMultiValue(extractNestedField(hit));
+            value = extractNestedField(hit);
         } else {
-            field = hit.field(fieldName);
+            DocumentField field = hit.field(fieldName);
             if (field != null) {
-                value = unwrapFieldsMultiValue(field.getValues());
+                value = field.getValues();
             }
         }
-        return value;
+        return unwrapFieldsMultiValue(value);
```

Comment:
This change is sliiightly different than the original version. I wouldn't change it unless you have a good reason for the change.